tests/python/gaia-ui-tests/gaiatest/apps/base.py

```diff
@@ -133,6 +133,17 @@ def a11y_select(self, match_string):
         self.accessibility.click(self.marionette.find_element(*_close_button_locator))
         self.wait_for_select_closed(*_close_button_locator)
 
+    def tap_element_from_system_app(self, element=None, add_statusbar_height=False):
+        # Workaround for bug 1109213, where tapping on the button inside the app itself
+        # makes Marionette spew out NoSuchWindowException errors
+        x = element.rect['x'] + element.rect['width']//2
+        y = element.rect['y'] + element.rect['height']//2
+        from gaiatest.apps.system.app import System
+        system = System(self.marionette)
+        if add_statusbar_height:
+          y = y + system.status_bar.height
+        system.tap(x, y)
+
```

Comment:
Just wondering, you don't have to switch to the root frame when you used this (is it something to do with it sitting in base.py)?  Because in cases where you replace the method, self.marionette.switch_to_frame() is always called.